src/main/java/fr/quatrevieux/araknemu/game/fight/castable/effect/buff/Buff.java

```diff
@@ -36,17 +36,23 @@
     final private ActiveFighter caster;
     final private PassiveFighter target;
     final private BuffHook hook;
+    final private boolean canBeDispelled;
 
     private int remainingTurns;
 
     public Buff(SpellEffect effect, Castable action, ActiveFighter caster, PassiveFighter target, BuffHook hook) {
+        this(effect, action, caster, target, hook, true);
+    }
+
+    public Buff(SpellEffect effect, Castable action, ActiveFighter caster, PassiveFighter target, BuffHook hook, boolean canBeDispelled) {
         this.effect = effect;
         this.action = action;
         this.caster = caster;
         this.target = target;
         this.hook = hook;
 
         this.remainingTurns = effect.duration();
+        this.canBeDispelled = canBeDispelled;
```

Comment:
Set this line after `this.hook = hook;` to group with other constructor parameter affectations